src/language-markdown/printer-markdown.js

```diff
@@ -200,11 +200,14 @@ function genericPrint(path, options, print) {
           return ["<", url, ">"];
         }
         case "[":
+          const encodedString = node.url
+            .replace(/</g, "%3C")
+            .replace(/>/g, "%3E");
```

Comment:
Move this logic to the `printUrl` function for supporting image and definition links. (Please forget adding tests!)

**Prettier pr-15400**
[Playground link](https://deploy-preview-15400--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBtANgSygawLoAUAFjDAA4DOSA9NQO4MB0A5hBM+nIwGYBO1BAJTc2AWgB8AIwCGvQYwA6UAIQZs+YqUo16TVu049+QkRAky5S0QAI1uPEmslyVWgzos2HLnwHCxUrIgADQgEGQwmNAUyKCyvBB0AAqyCDEo0uh00gCeMaGSvNJgOHAwAMrSALZwADLYcMjcmRRwBUUlZeVkxdjMyDC8AK5tIK1VmAPDoxR9nACKQxDwTS2jAFYUAB7lc3CLy41IzeitoQCOS-BJCZTIINIUolBwcAAm7yEgg9KYWFDMADCECqVWk90y6C+swBnAAgqReJhJENrnBePUXqtTqMSFV0AB1IiYeAUHpgODlNIkzAANxJOXuYAo+RAtJGAEkoB9YOUwEiInDueUYDlONiziAyAlWgSimR7tK4K1eLTGqFsCqYDdpMwwRLRj1eCr7mDeDg3okoF9pdgYATMG8YERkAAOAAMoV4cEumG9Or14OOa1CMGkkgdTpdSAATKEhq0ACrh9InSVwKqSd4fN61aQAoa6uAAMQgvDBpD6ENREBAAF860A)
<!-- prettier-ignore -->
```sh
--parser markdown
```

**Input:**
<!-- prettier-ignore -->
```markdown
[link](https://www.google.fr/()foo->bar).
![link](https://www.google.fr/()foo->bar)
- [link]: https://www.google.fr/()foo->bar
```

**Output:**
<!-- prettier-ignore -->
```markdown
[link](<https://www.google.fr/()foo-%3Ebar>).
![link](<https://www.google.fr/()foo->bar>)

- [link]: https://www.google.fr/()foo->bar

```

Reply:
Sorry, just wanted to confirm you don't want me to add tests?

Reply:
Ooops sorry!! Please **don't** forget adding tests!